File: BLogicSpring/src/main/java/ru/blogic/blogicspring/entity/document/Outgoing.java
package ru.blogic.blogicspring.entity.document;

import lombok.*;
import lombok.experimental.SuperBuilder;
import ru.blogic.blogicspring.entity.staff.Person;

import javax.persistence.*;

/**
 * Класс исходящего документа
 *
 * @author evaleev
 */
@Entity
@Table(name = "outgoing")
@Getter
@Setter
@ToString(callSuper = true)
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
public class Outgoing extends Document {

    /**
     * Поле адресата исходящего документа
     */
    @ManyToOne(cascade = CascadeType.PERSIST)
    @JoinColumn(name = "addressee_id")
    private Person addressee;
    /**
     * Поле метода доставки исходящего документа
     */
    @Enumerated(EnumType.STRING)
    @Column(name = "delivery_method")
    private DeliveryMethod deliveryMethod;

}

File: BLogicAngular/src/app/model/document/outgoing.ts
import {Document} from "./document";
import {Person} from "../person/person";

export class Outgoing extends Document {

  addressee: Person;
  deliveryMethod: string;

  constructor(id: string,
              name: string,
              text: string,
              registrationNumber: number,
              dateOfRegistration: Date,
              author: Person,
              addressee: Person,
              deliveryMethod: string) {
    super(id, name, text, registrationNumber, dateOfRegistration, author);
    this.addressee = addressee;
    this.deliveryMethod = deliveryMethod;
  }
}
